iceoryx_utils/include/iceoryx_utils/internal/concurrent/lockfree_queue/index_queue.hpp

```diff
@@ -95,6 +101,12 @@ class IndexQueue
     /// @return true if an index was obtained, false otherwise
     bool popIfFull(ValueType& index) noexcept;
 
+    /// @brief pop an index from the queue in FIFO order if the queue contains at least minSize indices
+    /// @param minSize minimum number of indices required in the queue to successfully obtain the first index
+    /// @param index that was obtained, undefined if false is returned
+    /// @return true if an index was obtained, false otherwise
+    bool popIfSizeIsAtLeast(uint64_t minSize, ValueType& index) noexcept;

```

Comment:
I think the method with return value `cxx::optional<ValueType>` functions is equally verbose but is safer and less prone to misuse compared to this one, is there a specific reason why this version exists or is exposed to the public at least?

Reply:
Yes, I wrote it as a version without optional and wanted to leave it to the user which one he prefers. It is an ongoing discussion, but at least for larger data types (complex constructors) the optional is slightly more heavyweight from what I know. 

My idea is to internally use the C-style version but I can make it inaccessible. Note that this class is in principle an internal one, but it is entirely possible to use this IndexQueue for specific tasks were it is required to manage some indices to an external buffer (could replace Loffli), so it should have safe interface.

Reply:
I think I will make the method private.